popgen/dunn.go

```diff
@@ -41,15 +41,10 @@ func Dunn(b *bed.Bed, aln []*fasta.Fasta, g []*Group) (float64, string) {
 //FindMaxIntra is a helper function of Dunn that calculates the Max pairwise sequence distance between two sequences of a multiFa alignment that are part of the same Group.
 func FindMaxIntra(subFa []*fasta.Fasta, g *Group, b *bed.Bed) int {
 	var answer int = 0
-	var group1index int
-	var group2index int
+	faMap := fasta.ToMap(subFa)
 	for i := 0; i < len(g.Members); i++ {
 		for j := i + 1; j < len(g.Members); j++ {
-			group1index = fasta.FindFaIndex(subFa, g.Members[i])
-			group2index = fasta.FindFaIndex(subFa, g.Members[j])
-			if group1index != -1 && group2index != -1 {
-				answer = numbers.Max(answer, dna.Dist(subFa[group1index].Seq, subFa[group2index].Seq))
-			}
+			answer = numbers.Max(answer, dna.Dist(faMap[g.Members[i]], faMap[g.Members[j]]))
```

Comment:
I think this code change looses the checks to make sure the sequence name was found.  I am not sure how dna.Dist performs if it gets nil for one of the seq.  Should this panic if the names are not found in the map (@rimangan)?

Reply:
I have added a check and panic if fastas not found. Riley, let me know if this should not panic.